wgengine/router/router_userspace_bsd.go

```diff
@@ -101,9 +102,42 @@ func inet(p netaddr.IPPrefix) string {
 	return "inet"
 }
 
+// See https://github.com/tailscale/tailscale/issues/1307#issuecomment-786045280
+// Remove all IPv6 entries.
+func (r *userspaceBSDRouter) modifiedConfigForFreeBSDBugWorkaround(cfg *Config) *Config {
+	n := cfg
```

Comment:
n := new(Config)

Reply:
I think this will break DNS on line 215, by not copying cfg.DNS.

Even if I copy all currently existing fields from the original cfg, we're just one CL away from someone adding another field that will be silently dropped on FreeBSD systems, at least as long as this code exists.

Searches implied that `n := cfg` would do a deep copy, but clearly not of slices or nested structs. A thread on #eng mentions https://github.com/tailscale/tailscale/blob/main/cmd/cloner/cloner.go as one possibility. Does that seem reasonable?

Reply:
cmd/cloner is fine, but I just looked at it seems like it is safe for you to mutate the passed in config. That's not documented right now (so it's not a current guarantee) but a few places of documentation could make that clear:

* local.go's `func routerConfig` (that the returned router.Config must not alias its `cfg` argument)
* wgengine/router/router.go's `Router.Set` method, to say that it owns the input.
*

Reply:
Went with Clone() as it seemed easier to revert later if/when we get IPv6 to actually work, without making new commitments about data structures.